src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.cs

```diff
@@ -788,33 +800,54 @@ private async ValueTask<bool> HandleSetColor(ClientPlayer sender, ColorType colo
             {
                 if (RequestedColorId.Any())
                 {
-                    var expected = RequestedColorId.Dequeue();
+                    var requested = RequestedColorId.Dequeue();
 
-                    for (var colorOffset = 0; colorOffset <= ColorsCount; colorOffset++)
+                    if (Game.IsColorUsed(color, this))
                     {
-                        var possibleColor = (ColorType)((byte)(expected + colorOffset) % ColorsCount);
-                        if (!Game.Players.Any(x => x.Character != null && x.Character != this && x.Character.PlayerInfo.CurrentOutfit.Color == possibleColor))
+                        if (await sender.Client.ReportCheatAsync(RpcCalls.SetColor, CheatCategory.ColorLimits, "Client selected a color that is already used"))
                         {
-                            expected = possibleColor;
-                            break;
+                            return false;
                         }
+                    }
+
+                    var startFrom = requested;
 
-                        if (colorOffset == ColorsCount)
+                    // Among Us wraps to the front if all colors between the requested and the final color are in use, check for this
+                    if (requested > color)
+                    {
+                        // Among Us wrapped, so check all colors between requested and the final color
+                        for (var c = (ColorType)requested; c < (ColorType)ColorsCount; c++)
```

Comment:
```suggestion
                        for (var c = requested; (byte)c < ColorsCount; c++)
```
`requested` is already a `ColorType` so that cast is unnecessary. And imo comparing bytes makes more sense